File: bindings/python/examples/secret_manager/generate_addresses.py
import os

from dotenv import load_dotenv
from iota_sdk import CoinType, MnemonicSecretManager, SecretManager

load_dotenv()

if 'MNEMONIC' not in os.environ:
    raise Exception(".env MNEMONIC is undefined, see .env.example")

# In this example we will create addresses from a mnemonic

secret_manager = SecretManager(MnemonicSecretManager(os.environ['MNEMONIC']))

# Generate public address with default account index and range.
addresses = secret_manager.generate_ed25519_addresses()

print('List of generated public addresses:', *addresses, sep='\n', end='\n\n')

addresses = secret_manager.generate_ed25519_addresses(
    coin_type=CoinType.SHIMMER,
    account_index=0,
    start=0,
    end=4,
    internal=False,
    bech32_hrp='rms')

print('List of generated public addresses:', *addresses, sep='\n', end='\n\n')

File: bindings/nodejs/examples/secret_manager/generate-addresses.ts
// Copyright 2021-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { SecretManager, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./secret_manager/generate-addresses.ts

// In this example we will create addresses from a mnemonic defined in .env
async function run() {
    initLogger();
    for (const envVar of ['MNEMONIC']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }
    try {
        const mnemonicSecretManager = {
            mnemonic: process.env.MNEMONIC as string,
        };

        const secretManager = new SecretManager(mnemonicSecretManager);

        // Generate public address with default account index and range.
        const default_addresses = await secretManager.generateEd25519Addresses(
            {},
        );
        console.log(
            'List of generated public addresses: ',
            default_addresses,
            '\n',
        );

        // Generate public address with custom account index and range.
        const address = await secretManager.generateEd25519Addresses({
            accountIndex: 0,
            range: {
                start: 0,
                end: 4,
            },
        });
        console.log('List of generated public addresses:', address, '\n');

        // Generate internal addresses with custom account index and range.
        const internalAddresses = await secretManager.generateEd25519Addresses({
            accountIndex: 0,
            range: {
                start: 0,
                end: 4,
            },
            options: { internal: true },
        });
        console.log(
            'List of generated internal addresses:',
            internalAddresses,
            '\n',
        );
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
